src/test/lib/ApiTest.cpp

```diff
@@ -5980,549 +5980,322 @@ QuicTestCredentialLoad(const QUIC_CREDENTIAL_CONFIG* Config)
     TEST_QUIC_SUCCEEDED(Configuration.LoadCredential(Config));
 }
 
+
+class QuicStorageSettingScopeGuard {
+public:
+    static
+    QuicStorageSettingScopeGuard Create(
+        _In_opt_ const char* StorageName) {
+        return QuicStorageSettingScopeGuard(StorageName);
+    }
+
+    QuicStorageSettingScopeGuard(const QuicStorageSettingScopeGuard&) = delete;
+    QuicStorageSettingScopeGuard& operator=(const QuicStorageSettingScopeGuard&) = delete;
+
+    QuicStorageSettingScopeGuard(
+        _In_ QuicStorageSettingScopeGuard&& Other) noexcept : m_Storage(Other.m_Storage) {
+        Other.m_Storage = nullptr;
+    }
+
+    QuicStorageSettingScopeGuard& operator=(QuicStorageSettingScopeGuard&& Other) = delete;
```

Comment:
nit: It is slightly strange to implement the move constructor but not the move assignment. This is typically shared with the move-constructor through a `swap` implementation, but to keep it short:

```suggestion
    QuicStorageSettingScopeGuard& operator=(QuicStorageSettingScopeGuard&& Other) {
        m_Storage = Other.m_Storage;
        Other.m_Storage = nullptr;
    }
```